tests/cpp_api_parser_test.py

```diff
@@ -41,11 +42,87 @@ def test_enable_if_transform():
         allow_paths=[re.escape(input_path)],
         disallow_namespaces=[re.compile("^std$")],
         compiler_flags=["-std=c++17"],
-        input_content=input_content)
+        input_content=input_content,
+    )
 
     output = api_parser.generate_output(config)
 
     entities = list(output["entities"].values())
     assert len(entities) == 1
     requires = entities[0].get("requires")
     assert requires
+
+
+def test_comment_strip():
+    # this test can also be used for trailing doc strings (though it isn't supported yet)
+    config = api_parser.Config(
+        input_path="a.cpp",
+        input_content="""
+//! This is a docstring.
+int var = 0;
+
+/// This is a docstring.
+int arr[1];
+
+// Do not show this.
+bool internal = true;  ///< This is a docstring.
+
+/** This is a docstring. */
+bool is_used = false;
+
+/*! This is a docstring. */
+bool is_useful = false;
+""",
+    )
+
+    output = api_parser.generate_output(config)
+    doc_strings = [
```

Comment:
I think it would be clearer to test each of these individually, so that there is just a single entity and you can just exactly check it.

Reply:
If I hadn't done it this way, then I never would've guessed that the `re.sub(brief|details, "", txt)` call needed the `re.MULTILINE` flag.

Are you ok with testing each regex pattern per test, instead of testing each supported command per test?

Reply:
I'm not sure I understand why this particular test structure helped spot that issue.  But I just meant that it would be clearer to have the test just generate one entity at a time, so that each test case is more isolated, at least for these comment syntax tests.

Reply:
I noticed that the re.sub() call was only replacing the first `\\brief` on the first line but not `\\details` on the third line of the same comment... The multiline flag fixed that.

I could split up this test into per comment syntax form. Initially, I thought this comment was for the other `test_function_fields()` test. 

Looking at the regex patterns, I don't know why you're looking for some fields like `checks`, `dchecks`, `schecks` (not really sure what they are supposed to represent). I get the `pre`, `post`, and `invariant` commands because those actually exist for Doxygen. The `error` (which isn't treated like an admonition) and `requires` (which seems equivalent to [`@concept`](https://doxygen.nl/manual/commands.html#cmdconcept)) commands seem to be something you added (they don't exist for Doxygen).

Reply:
Yes, checks, error, and requires are "custom" commands I added for tensorstore, along with corresponding custom sphinx docfields that I left out.  I agree they shouldn't be provided by default --- it would be good instead to have a mechanism for custom doxygen commands though.

Reply:
We can look into implementing something similar to [Doxygen's `ALIASES` setting](https://doxygen.nl/manual/config.html#cfg_aliases). That is [how breathe supports raw RST in Doxygen's XML output](https://breathe.readthedocs.io/en/latest/markups.html#aliases). Theoretically it can be used to add custom commands like the ones you're adding.

Reply:
I split up the comment stripping tests into a `class` of tests. With using `raw_comment`, the test for trailing docstring comment is passing. 🎉 

I could split up the test about function fields, but I think that can be done after the `_normalize_doc_comment()` function matures more (it is currently pretty limited).